### PR TITLE
Added method IsSimplicial for toric variety

### PR DESCRIPTION
**Pull request by HereAround**

### CODE REVIEW

**Comment by sebasguts** [comment]
This exists under the name `IsOrbifold`.

**Comment by HereAround** [comment]
Ok. Maybe this is still useful as convenience method?

**Comment by sebasguts** [comment]
Yeah, we can do that. We just need to be careful that the properties will be kept in sync.

**Comment by HereAround** [comment]
Indeed. The new commit should sync IsOrbifold and IsSimplicial.

**Comment by HereAround** [comment]
Ping

**Comment by sebasguts** [comment]
@mohamed-barakat Can I merge this?

**Comment by HereAround** [comment]
I just realized that this implementation causes a recursion depth trap, since IsSimplicial( variety ) calls IsSimplicial( variety ). The above commit should fix this.

File: ToricVarieties/gap/ToricVarieties.gi [hunk 1]
```diff
@@ -374,9 +374,28 @@ InstallMethod( IsOrbifold,
                [ IsFanRep ],
                
   function( variety )
+    local is_orbifold;
     
-    return IsSimplicial( FanOfVariety( variety ) );
```
[the remark this answers is not in the document]
Comment (HereAround) [reply ToricVarieties/gap/ToricVarieties.gi]: I am sorry, I do not understand the question. Can you please be more specific?

**Comment by HereAround** [comment]
I am still not sure, if I understand your question above completely. Still, to clear this PR up, I have now combined it all into a single commit.

Also, let me elaborate a bit more on the logic behind the code...

Here, we introduce the properties IsSimplicial and IsOrbifold for a toric variety. We do this merely for convenience - these are just different names for the very same property. Consequently, we need to synchronize IsSimplicial and IsOrbifold. 

This synchronization is realized as follows: Whenever the method IsOrbifold is being called, we first compute IsSimplicial( FanOfVariety( variety ) ). Originally, this value was return immediately. Now however, we remember this value by saving it in the variable is_orbifold. Next we set the value of the property IsSimplicial( variety ) to the value of is_orbifold. Finally, we return is_orbifold.

The implementation of IsSimplicial functions along the same lines.

Does this explain why "FanOfVariety" disappeared from the original line "return IsSimplicial( FanOfVariety( variety ) );"?

File: ToricVarieties/gap/ToricVarieties.gi (hunk 1 above)
[the remark this answers is not in the document]
Comment (HereAround) [reply ToricVarieties/gap/ToricVarieties.gi]: Ok, maybe I understand the question by now. I try to explain this it in detail below.

**Comment by mohamed-barakat** [comment]
What is the status of this PR?

File: ToricVarieties/gap/ToricVarieties.gi (hunk 1 above)
[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply ToricVarieties/gap/ToricVarieties.gi]: I would at least let one of them call the other. Why should we have the method twice?

[the remark this answers is not in the document]
Comment (HereAround) [reply ToricVarieties/gap/ToricVarieties.gi]: I agree - this should be better now.

[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply ToricVarieties/gap/ToricVarieties.gi]: I would rather leave the method for `IsOrbifold` untouched and use in the new method `IsSimplicial`.

[the remark this answers is not in the document]
Comment (HereAround) [reply ToricVarieties/gap/ToricVarieties.gi]: Ok. I have changed the commit such that indeed IsOrbifold is unchanged and IsSimplicial simply returns IsOrbifold( variety ).

**Pull request merged**